servers/display_server.cpp

```diff
@@ -1258,7 +1264,12 @@ bool DisplayServer::can_create_rendering_device() {
 			memdelete(rd);
 			rd = nullptr;
 			if (err == OK) {
+				// Creating a RenderingDevice is quite slow.
+				// Cache the result for future usage, so that it's much faster on subsequent calls.
+				created_rendering_device = RenderingDeviceCreationStatus::SUCCESS;
```

Comment:
Not your fault, but I just noticed this leaks memory. We need to free the RCD here before returning. Do you mind adding that to your PR?
```suggestion
				created_rendering_device = RenderingDeviceCreationStatus::SUCCESS;
				memdelete(rcd);
				rcd = nullptr;
```